test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -521,6 +521,19 @@ grpc_millis NowFromCycleCounter() {
   return grpc_timespec_to_millis_round_down(gpr_now(GPR_CLOCK_MONOTONIC));
 }
 
+// There can be milliseconds of difference between cycle time and time spec,
+// when the difference is in the order of a few seconds, we need to know that so
+// that our tests can account for the difference and reduce flakes due to the
+// clock difference.
+grpc_millis ClockDifference() {
+  gpr_cycle_counter now = gpr_get_cycle_counter();
+  grpc_millis cycle_time = grpc_cycle_counter_to_millis_round_up(now);
+  grpc_millis time_spec =
+      grpc_timespec_to_millis_round_down(gpr_now(GPR_CLOCK_MONOTONIC));
+  return (cycle_time >= time_spec ? cycle_time - time_spec
```

Comment:
This can just be `return std::abs(cycle_time - time_spec)`.

That having been said, do we actually want to adjust by the absolute value here?  We know that the tests are using cycle time, so if there's a discrepancy, C-core must be using timespec time.  So if cycle time is less than timespec time, then we know the test will be expecting a value lower than what C-core will be producing, and if cycle time is greater than timespec time, then we know that the test will be expecting a value greater than what C-core will be producing.  Can't we adjust in the right direction in each case?

Reply:
changed according to suggestions below